app/services/StorageService.service.js

```diff
@@ -0,0 +1,34 @@
+(function() {
+  'use strict';
+
+  class StorageService {
+
+    get(keyValue) {
+      return new Promise((resolve, reject) => {

```

Comment:
ok. wydaje mi sie, ze w tym miejscu tkwi cale zlo tej aplikacji: wszystko dziala u Ciebie dobrze, bo de facto Twoje promisy wywoluja sie synchronicznie. gdybysmy zmienili je na wywolania http albo cokolwiek innego moglibysmy miec pozniej problem (przynajmniej na poczatku dzialania aplikacji). 

sprobuj prosze ujac to w setTimeout(). czyli:
```javascript
setTimeout((resolve, reject) => {
  let videoArr = localStorage.getItem(keyValue);
  resolve(JSON.parse(videoArr));
  reject(new Error('This keyValue does not exist, try with different keyValue'));
}, 5000);
```